mysite/models/models.py

```diff
@@ -1885,6 +1913,20 @@ def _calculate_w3_tower_max_levels(self):
                     self.construction_buildings[towerName]['MaxLevel'] += 2 * self.atom_collider['Atoms']['Carbon - Wizard Maximizer']['Level']
                 except:
                     continue
+        
+        shrines = [shrineName for shrineName, shrineValuesDict in self.construction_buildings.items() if shrineValuesDict['Type'] == 'Shrine']
+        if sum(self.all_skills['Construction']) > 1500 and self.rift['SkillMastery']:
+            for shrineName in shrines:
+                try:
+                    self.construction_buildings[shrineName]['MaxLevel'] += 30
+                except:
+                    continue
+
+        if sum(self.all_skills['Construction']) > 500 and self.rift['SkillMastery']:
+            self.construction_buildings["Trapper Drone"]['MaxLevel'] += 35
+        
+        if sum(self.all_skills['Construction']) > 1000 and self.rift['SkillMastery']:
+            self.construction_buildings["Talent Book Library"]['MaxLevel'] += 100
```

Comment:
* Can these all 4 live under a single `if self.rift['SkillMastery']:` and also be in ascending requirement order? 500, 1000, 1500, and then 2500. 
* I'm critiquing myself here more than you, since you followed my lead. I should have assigned `towers` inside the `if` block. No point in creating that variable if it can't be used. Same can now be said for `shrines`.
* I'm also noticing I messed up on the original and made the check strictly `>` instead of `>=`. Could you update all 4? 
* Thank you for taking the time to add them all 👍 Clearly I was lazy in the past.